### PR TITLE
Add property datasets

### LINKED ISSUE #2: Get Data for Training Property Adaptors

### PR DESCRIPTION
**Pull request by Jacob-Chmura**
Close #2

### CODE REVIEW

File: README.md
Comment (Jacob-Chmura) [inline README.md:1]: Let's revert this change and put the prompting info directly into the datafile `README.md`

File: data/property_data/weight_data/data.csv
Comment (Jacob-Chmura) [inline data/property_data/weight_data/data.csv:1]: Given that this has _all_ the properties, and not just the _weight_, we should rename the directories to reflect that

File: data/property_data/weight_data/README.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+**Prompt**:
+
+<blockquote>
+
+1. In a single column, return 1000 distinct items
+1. Add another column that numerically categorizes each item according to it's size with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's physical phase with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's sentience with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's weight with respect to the reference points: rock, tree, ant, cat, chimp, man.
+1. Add another column that numerically categorizes each item according to it's rigidity with respect to the reference points: rock, tree, ant, cat, chimp, man.
```
Comment (Jacob-Chmura) [inline data/property_data/weight_data/README.md]: @jonahdvt Please update the reference points for the _non-sentience_ categories in accordance with how you prompted the model.

File: data/property_data/data.csv [hunk 2]
```diff
@@ -0,0 +1,906 @@
...
+Grass,1,3,1,1,1
+Rock,4,3,1,2,4
+Mountain,9,3,1,7,4
+River,9,2,1,7,1
+Ocean,9,2,1,7,1
+Star,9,4,1,7,5
+Moon,9,4,1,7,5
+Sun,9,4,1,7,5
+Cloud,9,1,1,7,1
+Rain,1,1,1,1,1
```
Comment (Jacob-Chmura) [inline data/property_data/data.csv]: @jonahdvt Why are these numerically encoded as 9 when there are only 5 reference points?

File: data/property_data/data.csv
Comment (Jacob-Chmura) [inline data/property_data/data.csv:1]: ### Distinct Items
@jonahdvt did you have a look at the data? I still see several duplicates. More importantly, you can see that, in some cases, the duplicate samples are classified differently:

foo

Comment (Jacob-Chmura) [inline data/property_data/data.csv:1]: ### Data Distribution
It's unlikely that the model would have by happen-chance, generated items that uniformly cover our reference points. But in case you haven't looked at the distribution, here is what it looks like:

foo

And perhaps the skew is more apparent in this chart:

foo

### Conclusion
I suggest we hand prompt for some of the missing categories, or we re-think our reference points to make them a bit easier to discriminate. In either case, we must investigate duplicates and _out-of-domain_ values like 7-9.

Comment (jonahdvt) [reply data/property_data/data.csv:1]: we can remove the duplicates. if anything the closeness in classification shows confidence in the model. it only differs very slightly so it shows that the data is classified pretty well. would be more concerning if they were vastly different

File: data/property_data/README.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+5. Add another column that numerically categorizes each item according to it's weight with respect to the reference points: watch, book, dumbbell, man, jeep, house, stadium.
+6. Add another column that numerically categorizes each item according to it's rigidity with respect to the reference points: water, skin, leather, wood, metal.
+
+Generate the data in CSV format. Write one item per line with the item and it's appropriate categories seperated by a commas.
+
+</blockquote>
+
+_Note_: the LLM had to be repeatedly prompted until about 1000 distinct items were generated. To maintain attention, the property attributes were generated in baches of 100 items.
+_Note_: some indices in the size category were registered as "out of distribution", as there were points 1-9. That is due to the fact that size was too challenging for the LLM to categorise, hence the categories of "smaller than ant" and "larger than stadium" were added, and the other categories were made "in between" reference points. THis results in 9 instead of 8 categories.
```
Comment (Jacob-Chmura) [inline data/property_data/README.md]: I would prefer if we just remap them to a common range. With this explanation alone, it's not super clear why the LLM generated numbers beyond the prescribed bins. Maybe I'm missing something, but I feel that having a consistent range is better for downstream perf anyway

**Comment by Jacob-Chmura** [comment]
@jonahdvt Let's try to precompute train/test data splits as seems to be standard when working with these hugging face libraries.
